### ISSUE TITLE
module diskdata fails

### ISSUE THREAD

**Issue by smoneck** [ultrabug/py3status#940]
Status bar gives

    diskdata: Command 'df' returned non-zero exit status 1
Syslog gives

    Instance `diskdata`, user method `space_and_io` failed (CommandError) py3.py line 889

* py3status version 3.6rc1 (python 3.6.1)
* 4.11.3-1-ARCH

**Comment by lasers** [comment]
The only thing I can find about error code is this. `$ info '(coreutils) df invocation'`
```
   An exit status of zero indicates success, and a nonzero value
indicates failure.  Failure includes the case where no output is
generated, so you can inspect the exit status of a command like ‘df -t
ext3 -t reiserfs DIR’ to test whether DIR is on a file system of type
‘ext3’ or ‘reiserfs’.
```
There are not much to do here as `df` came back with an error code and no output. I ran `diskdata` with`timeout_cache = 0` for days and can't trigger this.

If this turns out to be an occasional recurring behavior for you, maybe we can add `try/catch` the
`CommandError` exception with empty values to keep the loop going? 

If it's not an occasional recurring behavior for you, I think we should close this issue as it's working correctly and that there was an unexpected behavior coming from `df` at that time.

**Comment by SenchoPens** [comment]
I had this issue. I ran df and at first glance it worked (the output was fine), but then I noticed that the first line is `df: /mnt: Transport endpoint is not connected`, I just forgot to `umount` flash drive that I used before taking it out. So I ran `umount /mnt`, restarted i3 and diskdata began to work! 
TL;DR Run `df` and check carefully the output that it is producing.

**Comment by smoneck** [comment]
For me, the bug is no longer present. I think since a system upgrade lately ... Whatever it provoked. Thank you anyway for the effort! :-)

**Comment by lasers** [comment]
There were no issue with the script. Can somebody please close this? Many thanks.
